README.md

```diff
@@ -14,9 +14,22 @@ go generate
 ```
 
 ## Using the clients
-The clients are broken out into their own packages by name. Currently, `openapi-generator` produces mutilated method names like `GetCrmV3ObjectsContactsContactIdGetById`. This will need to be fixed in the `openapi-generator` project at some point.
+The clients are broken out into their own packages by name and categorized by API version.
 ```go
+import "github.com/clarkmcc/go-hubspot/generated/v3/contacts"
+
 client := contacts.NewAPIClient(contacts.NewConfiguration())
+request := client.BasicApi.Create(context.Background())
+
+input := contacts.SimplePublicObjectInput{
+  Properties: map[string]string{
+    "firstname": "Adrian", 
+    "lastname": "Mott",
+  },
+}
+
+request.SimplePublicObjectInput(input)
+request.Execute()
```

Comment:
This example actually doesn't work and it's actually a weird design decision from `openapi-generator`. If you print the `request` before and after adding the `input`, the `request` is unchanged.

```go
client := contacts.NewAPIClient(contacts.NewConfiguration())
request := client.BasicApi.Create(context.Background())
input := contacts.SimplePublicObjectInput{
	Properties: map[string]string{
		"firstname": "Adrian",
		"lastname":  "Mott",
	},
}

fmt.Printf("%+v\n", request)
request.SimplePublicObjectInput(input)
fmt.Printf("%+v\n", request)
```

It's interesting that `openapi-generator` generated this
```go
func (r ApiCreateRequest) SimplePublicObjectInput(simplePublicObjectInput SimplePublicObjectInput) ApiCreateRequest {
```
rather than this
```go
func (r *ApiCreateRequest) SimplePublicObjectInput(simplePublicObjectInput SimplePublicObjectInput) ApiCreateRequest {
```

Regardless, if you use method chaining, it should work as expected.

```go
request.SimplePublicObjectInput(input).Execute()
// or
request = request.SimplePublicObjectInput(input)
request.Execute()
```

Reply:
That's disappointing. Fixed!